text/0031-expanded_remove.md

```diff
@@ -0,0 +1,236 @@
+
+A better UX for removing data from qri should _clearly_ answer the common "uh oh" scenarios a user is likely to encounter:
+
+1. "I made a mistake while saving"
+2. "I want to delete a dataset I've pulled"
+3. "I want to undo a push"
+
+Theses are three distinct examples of removing versions, whole datasets, and remote replicas. This RFC proposes expanding the `remove` command to properly cover all of these common scenarios with the following changes:
+
+1. re-work the remove command to deal with all common cases of data removal. Formalize the interactions between these three removal scenarios 
+2. introduce _read-only_ and _read-write_ as terms to describe our present dataset acces model.
+2. improve help text and error messages, guiding users toward different uses of remove.
+3. Add an `amend` flag on the `save` command that _replaces_ the HEAD commit
+4. Add a `--remote` flag to `delete` for dropping datasets on remotes
+
+A well-documented desctrive command should be clearly labeled as the "opposite" of a constructive command and present warnings _before_ execution. Once executed, commands remove data should show as few caveat/errors/warnings as possible, or at least connect caveats to a clear next course of action. A feeling of "clean state" comes when vommands like `qri log`, `qri list` and `qri status` match user expectations after running a destructive command.
+
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+
+### "read-only" and "read-write"
+
+To get through this RFC we need language to distinguish between datasets the current user can and cannot edit, because it has an impact on the default behaviour of removing a dataset. **These terms should describe how qri currently works, not introduce new functionality**
+
+This RFC proposes distinguishing between _read-write_ and _read-only_ datasets like so:
+
+* datasets you create are _read-write_
+* all other users datasets are _read-only_
+* attempting to save to a read-only dataset creates a fork
+
+This proposes _qri already has an access control model_, we just need to make it explicit. Talking about our present system This gives us room for growth without introducing new terms. In the future what makes a dataset "readable" or "writable" will become more nuanced as we introduce collaboration features.
+
+### Expanding the Remove command
+
+This RFC sets aside the tight coupling between history and storage for another day, and instead focuses on fixing the role of remove by clarifying and expanding it's purpose.
+
+Here's the proposed help text for the new remove command:
+
+```
+$ qri remove --help
+Remove deletes datasets from qri.
+
+For read-only datasets you've pulled from others, Remove gets rid of a dataset 
+from your qri node. After running remove, qri will no longer list your dataset 
+as being available locally, and will free up storage space.
```

Comment:
Not sure if this is leftover from a previous revision, or new functionality, or just worded strangely. If I remove a dataset I've cloned from elsewhere, then run `remove`, are those blocks immediately freed up? Perhaps you meant, "and may eventually free up the storage space." Otherwise this seems both difficult to implement, and best left to the `storage` command.

Reply:
agree it's confusing, and will change to your proposed language.